inductiva/fluids/scenarios/coastal_area/random_bathymetry_utils.py

```diff
@@ -0,0 +1,251 @@
+    map_length = bathymetry_array.shape[0]
+    kernel_scale = inter_center_distance // 2
+
+    # Square Step, rows
+    for i in range(kernel_scale, map_length, inter_center_distance):
+        for j in range(0, map_length, inter_center_distance):
+            bathymetry_array[i,
+                             j] = kernel_avg(square_offsets, bathymetry_array,
+                                             i, j, kernel_scale)
+            bathymetry_array[i, j] += random.uniform(-noise_range, noise_range)
+
+    # Square Step, cols
+    for i in range(0, map_length, inter_center_distance):
+        for j in range(kernel_scale, map_length, inter_center_distance):
+            bathymetry_array[i,
+                             j] = kernel_avg(square_offsets, bathymetry_array,
+                                             i, j, kernel_scale)
+            bathymetry_array[i, j] += random.uniform(-noise_range, noise_range)
+
+
+def iterate_diamond_square(initial_condition: np.ndarray,
+                           initial_roughness: float = 1,
+                           roughness_factor: float = 0.5):
+    """Performs the diamond-square iteration."""
+
+    if initial_condition.shape[0] != initial_condition.shape[1]:
+        logging.error(
+            'The initial array must be a square. The current shape is: %f x %f',
+            initial_condition.shape[0], initial_condition.shape[1])
+
+    logging.info(
+        'Generating the bathymetry with the Diamond-Square Algorithm...')
+
+    map_length = initial_condition.shape[0]
+    inter_center_distance = map_length - 1
+    roughness = initial_roughness
+    random_bathymetry = initial_condition
+
+    while inter_center_distance > 1:
+
+        diamond_step(random_bathymetry, inter_center_distance, roughness)
+        square_step(random_bathymetry, inter_center_distance, roughness)
+
+        # Go to a finer grid, and decrease the amount of noise
+        inter_center_distance //= 2
+        roughness *= roughness_factor
+
+    return random_bathymetry
+
+
+def generate_inclined_bed_initial_condition(size, max_depth=10):
+    """Creates initial conditions for an inclined bed type of profile.
+
+    This is done by setting the left corners of the map lower than the right
+    corners. There is some randomness involved to allow for situations where
+    the bed is bit rounder,e.g. with more mass on one of the right corners than
+    the other."""
+
+    # Start with a flat base shape, all at zero level.
+    inclined_bed = np.zeros(shape=(size, size))
+
+    # Define left corners underwater
+    inclined_bed[0, 0] = random.uniform(0, max_depth)
+    inclined_bed[0, size - 1] = random.uniform(0, max_depth)
+
+    # define right corners above water
+    inclined_bed[size - 1, 0] = random.uniform(-max_depth, 0)
+    inclined_bed[size - 1, size - 1] = random.uniform(-max_depth, 0)
+
+    return inclined_bed
+
+
+def create_random_bathymetry(size: int,
```

Comment:
I was trying to run the code on my side and it seems that I can pass all sizes possible. Shouldn't the sizes be fixed for 2^k+1?

The bathymetry is not good for `sizes !=2^k+1`.

Reply:
This is a function in a utilities file, that (in principle) shouldn't be used directly by users, so I wouldn't bother much with making sure that this size is some 2^k + 1. I can always add a check, though. WDYT?

Reply:
A check is enough!